rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -134,6 +134,11 @@ Node::create_publisher(
       shared_publish_callback,
       intra_process_publisher_handle);
   }
+  if (rmw_trigger_guard_condition(notify_guard_condition_) != RMW_RET_OK) {
+    throw std::runtime_error(
+            std::string(
+              "Failed to notify waitset of change to node state: ") + rmw_get_error_string());
```

Comment:
That's a good idea. I changed the error messages to the format "Failed to notify waitset on X creation" in ddc6609, where X is the type of entity added to the node.